learn/advanced/working_with_dates.md

```diff
@@ -0,0 +1,118 @@
+
+## Preparing your documents
+
+To filter and sort search results chronologically, your documents must have at least one numeric field containing a [UNIX timestamp](https://kb.narrative.io/what-is-unix-time).
+
+As an example, consider a database of video games. In this dataset, the release year is formatted as a timestamp:
+
+```json
+[
+  {
+    "id": 0,
+    "title": "Return of the Obra Dinn",
+    "genre": "adventure",
+    "release_timestamp": 1538949600
+  },
+  {
+    "id": 1,
+    "title": "The Excavation of Hob's Barrow",
+    "genre": "adventure",
+    "release_timestamp": 1664316000
+  },
+  {
+    "id": 2,
+    "title": "Bayonetta 2",
+    "genre": "action",
+    "release_timestamp": 1411164000
+  }
+]
+```
+
+If your chronological data is expressed in another format, like [ISO 8601](https://www.iso.org/iso-8601-date-and-time-format.html), you must convert it to a numeric value before indexing it with Meilisearch.
+
+Most programming languages have built-in tools to help you with this process. The JavaScript example below converts a game's release date, `"2018-10-18"`, to a numeric timestamp:
+
+```js
+let game = {
+  "id": 0,
+  "title": "Return of the Obra Dinn",
+  "genre": "adventure",
+  "release_date": "2018-10-18T00:00Z"
+};
+
+const timestamp = Date.parse(game.year);
+
+game = {
+  "id": 0,
+  "title": "Return of the Obra Dinn",
+  "genre": "adventure",
+  "release_date": "2018-10-18T00:00Z",
+  "release_timestamp": timestamp
+};
+```
+
+:::tip
+When preparing your dataset, it can be useful to leave the original date and time fields in your documents intact. In the example above, we keep the `release_date` field because it is more readable than the raw `timestamp`.
+:::
+
+After adding a numeric timestamp to all documents, [index your data](/reference/api/documents.md#add-or-replace-documents) as usual. See below for an example using our nonexistent video games dataset.
+
+```sh
+curl \
+  -X POST 'http://localhost:7700/indexes/games/documents' \
+  -H 'Content-Type: application/json' \
+  --data-binary @games.json
+```
+
+## Filtering by timestamp
+
+To filter search results based on their timestamp, add your document's timestamp field to the list of [`filterableAttributes`](/reference/api/settings.md#update-filterable-attributes):
+
+```sh
+curl \
+  -X PUT 'http://localhost:7700/indexes/games/settings/filterable-attributes' \
+  -H 'Content-Type: application/json' \
+  --data-binary '[
+    "release_timestamp"
+  ]'
+```
+
+Once you have configured `filterableAttributes`, you can filter search results by date. The following query only returns games released between 2018 and 2022:
+
+```sh
+curl \
+  -X POST 'http://localhost:7700/indexes/games/search' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "q": "",
+    "filter": "release_timestamp >= 1514761200 AND release_timestamp < 1672527600"
+  }'
+```
```

Comment:
Maybe an example in JS would be more convenient to read than curl. I mean, maybe we could have a code part ([with `Date`](https://stackoverflow.com/questions/9873197/how-to-convert-date-to-timestamp)?) where we transform `2018` and `2022` into a timestamp. It's easier to understand we are searching between 2018 and 2022 with the date written than seeing `1514761200` and `1672527600` and guessing it means `2018` and `2022`

@bidoubiwa WDYT? Maybe it's not performant to do this?

Reply:
I agree that javascript is easier to read; the issue is that we need to import meilisearch-js.
Are you thinking about something like that: 

```js
const { MeiliSearch } = require('meilisearch')
const client = new MeiliSearch({ host: '...' })

let documents = [ { id: 1, release_date: "2018-10-18T00:00Z" },  {}] // etc

documents = documents.map(document => ({
  ...document,
  release_timestamp: Date.parse(document.release_date) / 1000,
}))
await client.index('games').addDocuments(documents)

// wait for tasks?

const searchResults = client.index('games').search('', {
  filters: 'release_timestamp >= 1514761200 AND release_timestamp < 1672527600',
})  // Where 1514761200 is Sunday, December 31, 2017  and 1514761200 is Saturday, December 31, 2022

```

My bad if I misunderstood. The code can be made more readable btw.

Reply:
That's oke for me!